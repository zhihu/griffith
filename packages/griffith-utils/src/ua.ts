export const isMobile = /iPhone|iPad|iPod|Android/i.test(navigator.userAgent)

export const isAndroid = /(android)/i.test(navigator.userAgent)

export const isSafari = /^((?!chrome|android).)*safari/i.test(
  navigator.userAgent
)

export default {
  isMobile,
  isAndroid,
  isSafari,
}
