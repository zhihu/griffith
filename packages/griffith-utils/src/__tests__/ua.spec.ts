/**
 * @jest-environment jsdom
 */

import ua, {parseUA} from '../ua'

test('ua', () => {
  expect(ua).toMatchInlineSnapshot(`
    Object {
      "isAndroid": false,
      "isIE": false,
      "isMobile": false,
      "isSafari": false,
    }
  `)
})

test('parse', () => {
  expect(
    parseUA(
      'Mozilla/5.0 (Linux; Android 5.0; SM-G900P Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Mobile Safari/537.36'
    )
  ).toMatchInlineSnapshot(`
    Object {
      "isAndroid": true,
      "isIE": false,
      "isMobile": true,
      "isSafari": false,
    }
  `)
})
