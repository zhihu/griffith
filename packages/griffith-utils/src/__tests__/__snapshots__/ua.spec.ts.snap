// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse 1`] = `
{
  "isAndroid": true,
  "isIE": false,
  "isMobile": true,
  "isSafari": false,
}
`;

exports[`ua 1`] = `
{
  "isAndroid": false,
  "isIE": false,
  "isMobile": false,
  "isSafari": false,
}
`;
